### PR TITLE
[DEV-7132] Obligated Amount Burnup Chart Redline

### PR DESCRIPTION
**Pull request by afnazu**
**High level description:**

Add line representing agency budget per fiscal year and shaded area representing difference between agency budget and most recent submission period obligation per fiscal year

**JIRA Ticket:**
[DEV-7132](https://federal-spending-transparency.atlassian.net/browse/DEV-7132)

**Mockup:**
[Mock](https://bahdigital.invisionapp.com/d/main#/console/14222398/296055430/preview)

The following are ALL required for the PR to be merged:

Author:
- [x] Linked to this PR in JIRA ticket
- [x] Scheduled demo including Design/Testing/Front-end OR Provided instructions for testing in JIRA and PR `if applicable`
- [x] Verified cross-browser compatibility: Chrome, Safari, Firefox, Internet Explorer, Edge
- [x] Verified mobile/tablet/desktop/monitor responsiveness
- [x] Verified that this PR does not create any *new* accessibility issues (via Axe Chrome extension)
- [n/a] Added Unit Tests for helper functions, reducers, models and Container/Component Interactivity Expectations `if applicable` [React Testing Library](react-testing-library.md)
- [n/a] [API contract](https://github.com/fedspendingtransparency/usaspending-api/tree/dev/usaspending_api/api_contracts) updated `if applicable`
- [n/a] [Component Library Integration Status Report](https://github.com/fedspendingtransparency/data-act-documentation/blob/data-transparency-ui/frontend_apps/component-library-integration-status.md) updated `if applicable`

Reviewer(s):
- [x] Design review complete `if applicable`
- [n/a] [API #1234](https://github.com/fedspendingtransparency/usaspending-api/pull/1234) merged concurrently `if applicable`
- [ ] Code review complete

### CODE REVIEW

File: src/js/components/agencyV2/visualizations/totalObligationsOverTime/AgencyBudgetLine.jsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    xDomain,
+    yScale,
+    height,
+    agencyBudget,
+    todaysDate,
+    padding
+}) => {
+    const [show, setShow] = useState(false);
+
+    useEffect(() => {
```
Comment (jonhill13) [inline src/js/components/agencyV2/visualizations/totalObligationsOverTime/AgencyBudgetLine.jsx:30]: We only want to show the agency budget during the current budget year?

File: src/js/components/agencyV2/visualizations/totalObligationsOverTime/AgencyBudgetLine.jsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    return (
+        <g>
+            {show && <line
+                tabIndex="0"
+                className="total-budget-line"
+                x1={padding.left}
+                x2={xScale(todaysDate) + padding.left}
+                y1={height - yScale(agencyBudget - padding.top)}
+                y2={height - yScale(agencyBudget - padding.top)} />}
+            {show && <rect className="total-budget-difference" x={padding.left} y={height - yScale(agencyBudget - padding.top)} width={xScale(todaysDate)} height={yScale(agencyBudget) - yScale(data[data.length - 1].obligated) - padding.bottom} />}
```
Comment (jonhill13) [inline src/js/components/agencyV2/visualizations/totalObligationsOverTime/AgencyBudgetLine.jsx:48]: For the `y` property of the rectangle, we want scale the `agencyBudget`. Our `yScale`'s domain is based off of monetary values from the api, and its range is based off our true graphing area with the padding removed.  We want to pass monetary values to the `yScale` which will return the y-coordinate. We remove the padding after we scale the monetary value.
 
`y={height - yScale(agencyBudget) - padding.top}`

Comment (jonhill13) [reply src/js/components/agencyV2/visualizations/totalObligationsOverTime/AgencyBudgetLine.jsx:48]: We want to do the same thing for the `y1` and `y2` properties of the `line`

```
y1={height - yScale(agencyBudget) - padding.top}
y2={height - yScale(agencyBudget) - padding.top}
```

**Review by jonhill13 — CHANGES_REQUESTED** [review]

File: src/js/components/agencyV2/visualizations/totalObligationsOverTime/AgencyBudgetLine.jsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        }
+    }, [xScale, xDomain, todaysDate]);
+
+    useEffect(() => {
+        if (xScale && yScale && agencyBudget) {
+            setLineData(
+                {
+                    x1: padding.left,
+                    x2: show ? xScale(todaysDate) + padding.left : width - padding.left,
+                    y1: height - yScale(agencyBudget) - padding.top
```
Comment (jonhill13) [inline src/js/components/agencyV2/visualizations/totalObligationsOverTime/AgencyBudgetLine.jsx:55]: can migrate this `y1` to it's own variable and check if it is `isNaN` to remove the console errors?

```
const y = height - yScale(agencyBudget) - padding.top;
```

then in the object

```
y1: isNaN(y1) ? 0 : y1
```

**Review by jonhill13 — CHANGES_REQUESTED** [review]

File: src/js/components/agencyV2/visualizations/totalObligationsOverTime/AgencyBudgetLine.jsx (hunk 3 above)
Comment (afnazu) [reply src/js/components/agencyV2/visualizations/totalObligationsOverTime/AgencyBudgetLine.jsx]: done!

**Review by jonhill13 — APPROVED** [review]

**Pull request merged**